asminc/ctypetable.inc

```diff
@@ -0,0 +1,48 @@
+; ctypetable.inc
+;
+; This file is part of
+; cc65 - a freeware C compiler for 6502 based systems
+;
+; https://github.com/cc65/cc65
+;
+; See "LICENSE" file for legal information.
+;
+; Data covering all possible combinations of character flags for target specific definition
+;
+
+.include	"ctype.inc"
+.export		__ctype
+
+; This data is a table of possible ctype combinations, possible during 
+
+.rodata
+__ctype:
+ct_none:				.byte CT_NONE
+ct_lower:				.byte CT_LOWER
+ct_upper:				.byte CT_UPPER
+ct_digit_xdigit:		.byte CT_DIGIT | CT_XDIGIT
+ct_lower_xdigit:		.byte CT_LOWER | CT_XDIGIT
+ct_upper_xdigit:		.byte CT_UPPER | CT_XDIGIT
+ct_ctrl:				.byte CT_CTRL
+ct_ws:					.byte CT_OTHER_WS
+ct_ctrl_ws:				.byte CT_CTRL | CT_OTHER_WS
+ct_space_spacetab:		.byte CT_SPACE | CT_SPACE_TAB
+ct_ctrl_ws_spacetab:	.byte CT_CTRL | CT_OTHER_WS | CT_SPACE_TAB
```

Comment:
Looking at the isspace / isblank code, do we need CT_OTHER_WS?
As the only place used is on "isspace", and there is used as (CT_SPACE | CT_OTHER_WS), imho, one flag should be enough.

Reply:
Seems that you are right.
I would change `CT_OTHER_WS` to 'CT_WHITESPACE' and remove the 'CT_SPACE' definition.
Comments?

Reply:
Had a second look. As this change wouldn't save a single byte (all entries in the ctype-table would be still in use) and `CT_CTRL_SPACE` and `CT_NOT_PUNCT` definition use `CT_SPACE` (in `isgraph()` and `ispunct()`) but not `CT_OTHER_WS` I won't touch the current state.